packages/google_maps_flutter/google_maps_flutter_web/lib/src/convert.dart

```diff
@@ -289,21 +289,32 @@ gmaps.Size? _gmSizeFromIconConfig(List<Object?> iconConfig, int sizeIndex) {
       );
     }
   }
+
   return size;
 }
 
-/// Sets the size of the Google Maps icon.
-void _setIconSize({
-  required Size size,
-  required gmaps.Icon icon,
+// Sets the size and style of the [icon] element.
+void _setIconStyle({
+  required web.Element icon,
+  required gmaps.Size? size,
+  required double? opacity,
+  required bool? isVisible,
 }) {
-  final gmaps.Size gmapsSize = gmaps.Size(size.width, size.height);
-  icon.size = gmapsSize;
-  icon.scaledSize = gmapsSize;
+  icon.setAttribute(
+    'style',
+    <String>[
+      if (size != null) ...<String>[
+        'width: ${size.width}px;',
+        'height: ${size.height}px;',
```

Comment:
I suspect the change in the test to ignore the decimals can be fixed here as well, by being more explicit with the conversion of `size.width`? Right now I think it's a `num`, so we could maybe call [`toStringAsFixed(1)`](https://api.flutter.dev/flutter/dart-core/num/toStringAsFixed.html), so this always renders the `num` with 1 decimal regardless of the compilation target?

Reply:
Great suggestion, thanks! That's a better solution than ignoring zeros 👍